src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -365,6 +364,10 @@ private void SetTransferMode(TransferMode transferMode)
                 Output.TryComplete(ex);
                 _logger.ErrorReceiving(_connectionId, ex);
             }
+            finally
+            {
+                _httpClient.Dispose();
```

Comment:
How is the client disposed when the connection is never started? I think it won't be...

Reply:
Interesting point, I don't think it ever was before?

Reply:
It did - this is why we had it in the `DisposeAsync`

Reply:
Why would you call `DisposeAsync` when the connection isn't started

Reply:
I don't think it matters (maybe the app has some additional logic between creating and starting the connection to decide) - we should dispose properly when the user called `DisposeAsync()`...

Reply:
So now we call Dispose in 3 places!? Can you write a comment saying why?